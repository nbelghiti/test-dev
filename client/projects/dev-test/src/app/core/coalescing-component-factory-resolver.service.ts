/* tslint:disable */
/**
 * See https://github.com/jonrimmer/angular-coalescing-component-factory-resolver/blob/master/src/app/coalescing-component-factory-resolver.service.ts
 * Discussion & bug ticket: https://github.com/angular/angular/issues/14324#issuecomment-481898762
 *
 * Entry components are only available within a lazy loaded module, breaking dynamic generation (e.g. dialogs). This monkey patch enables resolving entry components from lazy loaded modules as well.
 *
 *
 * Provide in root module:
 *
 * @NgModule({
 *   providers: [CoalescingComponentFactoryResolver]
 * })
 * class AppModule {
 *   constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
 *     coalescingResolver.init();
 *   }
 * }
 *
 *
 * Use in lazy loaded module:
 *
 *
 * @NgModule({})
 * export class LazyModule {
 *   constructor(
 *     coalescingResolver: CoalescingComponentFactoryResolver,
 *     localResolver: ComponentFactoryResolver
 *   ) {
 *     coalescingResolver.registerResolver(localResolver);
 *   }
 * }
 *
 *
 * How it works: When initialised, it injects the root app's ComponentFactoryResolver and monkey patches the resolveComponentFactory method
 * to call its own implementation. This implementation first tries resolving the component factory on all the registered lazy-module resolvers,
 * then falls back to the root app resolver (there's a bit of extra logic to avoid cyclic calls, but that's the gist).
 */
/* tslint:enable */

import {
	ComponentFactory,
	ComponentFactoryResolver,
	Injectable,
	Type
} from '@angular/core';

@Injectable()
export class CoalescingComponentFactoryResolver extends ComponentFactoryResolver {
	private rootResolve: (component: Type<any>) => ComponentFactory<any>;

	private inCall = false;

	private readonly resolvers = new Map<
		ComponentFactoryResolver,
		(component: Type<any>) => ComponentFactory<any>
	>();

	constructor(private readonly rootResolver: ComponentFactoryResolver) {
		super();
	}

	init() {
		this.rootResolve = this.rootResolver.resolveComponentFactory;
		this.rootResolver.resolveComponentFactory = this.resolveComponentFactory;
	}

	public registerResolver(resolver: ComponentFactoryResolver) {
		const original = resolver.resolveComponentFactory;
		this.resolvers.set(resolver, original);
	}

	public resolveComponentFactory = <T>(
		component: Type<T>
	): ComponentFactory<T> => {
		// Prevents cyclic calls.
		if (this.inCall) {
			return null;
		}

		this.inCall = true;
		try {
			const result = this.resolveInternal(component);
			return result;
		} finally {
			this.inCall = false;
		}
	}

	private resolveInternal = <T>(component: Type<T>): ComponentFactory<T> => {
		for (const [resolver, fn] of Array.from(this.resolvers.entries())) {
			try {
				const factory = fn.call(resolver, component);
				if (factory) {
					return factory;
				}
			} catch { }
		}

		return this.rootResolve.call(this.rootResolver, component);
	}
}
