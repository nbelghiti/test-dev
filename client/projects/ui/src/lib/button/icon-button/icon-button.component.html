<button
	*ngIf="!link; else isLink"
	class="c-icon-button"
	[title]="buttonTitle"
	[ngClass]="{
		'c-icon-button--secondary': type === ButtonType.secondary,
		'c-icon-button--tertiary': type === ButtonType.tertiary,
		'c-icon-button--border': border,
		'c-icon-button--fluid': fluid
	}"
	(click)="clicked.emit($event)"
>
	<ng-container [ngTemplateOutlet]="iconButtonContent"></ng-container>
</button>

<ng-template #isLink>
	<a
		[routerLink]="link"
		class="c-icon-button"
		[ngClass]="{
			'c-icon-button--secondary': type === ButtonType.secondary,
			'c-icon-button--tertiary': type === ButtonType.tertiary,
			'c-icon-button--border': border,
			'c-icon-button--fluid': fluid
		}"
	>
		<ng-container [ngTemplateOutlet]="iconButtonContent"></ng-container>
	</a>
</ng-template>

<ng-template #iconButtonContent>
	<ng-content></ng-content>
</ng-template>
