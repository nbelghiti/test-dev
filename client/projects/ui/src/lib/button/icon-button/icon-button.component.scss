// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$icon-button-size-base: 4rem;
$icon-button-size-small: 2.8rem;

// =============================================================================
// :: Icon button
// =============================================================================
.c-icon-button {
	position: relative;

	display: inline-flex;
	justify-content: center;
	align-items: center;

	width: $icon-button-size-base;
	height: $icon-button-size-base;

	text-decoration: none;

	border: $border-width-base solid transparent;
	color: $primary-color;

	transition:
		color $animation-base,
		border-color $animation-base;

	cursor: pointer;

	&:hover {
		border: $border-width-base solid $primary-color;
	}

	&:focus {
		border: $border-width-base dashed $primary-color;
	}
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon-button--border {
	border: $border-width-base solid $background-color-orange-78;
}

.c-icon-button--secondary,
.c-icon-button--tertiary {
	color: $text-color-grey-46;

	&.c-icon-button--border {
		border: $border-width-base solid $border-color-grey-86;
	}

	&:hover {
		color: $text-color-base;
		border-color: $text-color-base;
	}
}

.c-icon-button--tertiary {
	width: $icon-button-size-small;
	height: $icon-button-size-small;
}

.c-icon-button--fluid {
	width: 100%;
	height: 100%;

	color: $text-color-grey-46;
	border: none;

	&:hover,
	&:focus {
		border: none;
		color: $primary-color;
	}
}
