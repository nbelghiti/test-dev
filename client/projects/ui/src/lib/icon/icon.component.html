<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
