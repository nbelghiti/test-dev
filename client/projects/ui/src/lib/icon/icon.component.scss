// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
