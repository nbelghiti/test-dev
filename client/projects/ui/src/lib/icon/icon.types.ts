export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
