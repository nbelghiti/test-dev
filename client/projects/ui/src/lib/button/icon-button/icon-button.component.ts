import { Component, EventEmitter, Input, Output } from '@angular/core';

import { ButtonType } from '../button.types';

@Component({
	selector: 'ui-icon-button',
	templateUrl: './icon-button.component.html',
	styleUrls: ['./icon-button.component.scss']
})

export class IconButtonComponent {
	@Input() public type: ButtonType = ButtonType.primary;
	@Input() public link: string;
	@Input() public buttonTitle: string = '';
	@Input() public fluid: boolean = false;
	@Input() public border: boolean = true;

	@Output() public clicked: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();

	public ButtonType: typeof ButtonType = ButtonType;
}
