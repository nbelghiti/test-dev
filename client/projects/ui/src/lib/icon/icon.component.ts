import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
