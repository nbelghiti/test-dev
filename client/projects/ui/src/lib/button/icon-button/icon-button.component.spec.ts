// TODO: Add snapshot
import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { IconComponent } from '../../icon/icon.component';

import { IconButtonComponent } from './icon-button.component';

describe('IconButtonComponent', () => {
	let component: IconButtonComponent;
	let fixture: ComponentFixture<IconButtonComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [ RouterTestingModule ],
			declarations: [ IconButtonComponent, IconComponent ]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(IconButtonComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
