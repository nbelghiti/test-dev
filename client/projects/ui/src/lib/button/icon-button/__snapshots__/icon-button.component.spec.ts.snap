// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonComponent Should render correctly 1`] = `
<ui-icon-button
  ButtonType={[Function Object]}
  border={[Function Boolean]}
  buttonTitle=""
  clicked={[Function EventEmitter]}
  fluid="false"
  type=""
>
  <button
    class="c-icon-button c-icon-button--border"
    ng-reflect-klass="c-icon-button"
    ng-reflect-ng-class="[object Object]"
    title=""
  >
    
    
  </button>
</ui-icon-button>
`;
